### ISSUE TITLE
Support HLS in VideoUrlSource (to enable using the url for video download)

### ISSUE THREAD

**Issue by paaspaas00** [futo-org/grayjay-android#393]
Support HLS in VideoUrlSource (to enable using the url for video download), some platforms don't have direct links for mp4 files

**Comment by Kelvin-FUTO** [comment]
You should not be using VideoUrlSource for HLS sources, instead use HLSManifestSource which looks very similar.

VideoUrlSource effectively means a source that has video/audio files directly available, and thus has completely different logic to handle playback, download, casting, and so on.

I recommend you check out the docs:  
**https://github.com/futo-org/grayjay-android/blob/master/docs/Content%20Types.md**

If you use the correct types everything that is supported should automatically be supported.

**Comment by Kelvin-FUTO** [comment]
I don't believe we use and VideoUrlSource for HLS, because that would mean it wouldn't work at all.
You might be confusing Odysee, which uses both HLSSource and VideoUrlSource as it uses different content for different videos, but should never use VideoUrlSource for a HLS manifest, as that would not work in Grayjay.

HLSSource = HLSManifestSource, sorry for the confusion, We call it something else in the backend, we likely will add an alias in the future.
